packages/design-system/src/components/Tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,293 @@
+
+    this.state = { active: false };
+  }
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleClickOutside.bind(this));
+    document.addEventListener('keydown', this.handleEscapeKey.bind(this));
+    this.popper = createPopper(this.triggerElement, this.tooltipElement, {
+      placement: this.props.placement,
+      modifiers: [
+        {
+          name: 'offset',
+          options: { offset: this.props.offset },
+        },
+      ],
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside.bind(this));
+    document.removeEventListener('keydown', this.handleEscapeKey.bind(this));
+    this.popper.destroy();
+  }
+
+  handleClickOutside(event) {
+    // Closes click only tooltips when mouse clicks outside of tooltip container element
+    if (this.state.active && this.props.dialog) {
+      const clickedTooltip = this.tooltipElement && this.tooltipElement.contains(event.target);
+      if (!clickedTooltip) {
+        this.setTooltipActive(false);
+      }
+    }
+  }
+
+  handleEscapeKey(e) {
+    // Closes interactive and click only tooltips when ESC key is pressed
+    const ESCAPE_KEY = 27;
+    if (this.state.active && e.keyCode === ESCAPE_KEY) {
+      if (this.props.interactive || this.props.dialog) {
+        this.setTooltipActive(false);
+      }
+    }
+  }
+
+  setTooltipActive(active) {
+    this.setState({ active }, () => {
+      this.popper.forceUpdate();
+    });
+  }
+
+  handleBlur() {
+    // Hide tooltips when blurring away from the trigger or tooltip body
+    setTimeout(() => {
+      const focusedInsideTrigger =
+        this.triggerElement && this.triggerElement.contains(document.activeElement);
+      const focusedInsideTooltip =
+        this.tooltipElement && this.tooltipElement.contains(document.activeElement);
+      if (!focusedInsideTrigger && !focusedInsideTooltip) {
+        this.setTooltipActive(false);
+      }
+    }, 10);
+  }
+
+  triggerComponentType() {
+    let component = this.props.triggerComponent;
+    if (component === 'button' && this.props.triggerHref) {
+      // If `href` is provided and a custom component is not, we render `<a>` instead
+      component = 'a';
+    }
+    return component;
+  }
+
+  renderTrigger() {
+    const {
+      dialog,
+      ariaLabel,
+      triggerActiveClassName,
+      triggerClassName,
+      triggerContent,
+      triggerHref,
+      triggerId,
+    } = this.props;
+
+    const TriggerComponent = this.triggerComponentType();
+    const triggerClasses = classNames('ds-c-tooltip__trigger', 'ds-base', triggerClassName, {
+      [triggerActiveClassName]: this.state.active,
+      'ds-c-tooltip__trigger--click-only-active': this.props.dialog && this.state.active,
+    });
+
+    return (
+      <TriggerComponent
+        id={triggerId}
+        type={TriggerComponent === 'button' ? 'button' : undefined}
+        onTouchStart={() => this.setTooltipActive(!this.state.active)}
```

Comment:
Nice use of `onTouchStart`, first time to see this event. [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/ontouchstart#Browser_compatibility) list this as an experimental technology, I'm curious to continue trying this feature out on various browsers.